### ISSUE TITLE
Possible to bump NPM Version?

### ISSUE THREAD

**Issue by foxmicha** [anothrNick/github-tag-action#20]
Hey there, currently loving the Action for bumping tags. This is more of a feature request / question than it is an issue, so feel free to close it and direct me to a better tracking place if one exists.

Is there a clean way to bump the semver version of an NPM package using the same `#major`, `#minor`, `#patch` keywords from within this Action? Should this be done in an additional step using the `setup-node` Action?

Any insight is great. Thanks!

**Comment by anothrNick** [comment]
The tag created within this action is set as an output, so it is available to other actions further down the workflow chain. 

You'd want another step in your workflow to bump/release the version to npm using the latest tag. I'd have to look into it, but I would imagine an action to release to npm already exists.

**Comment by anothrNick** [comment]
After looking around a bit, I wonder if you can do something like this:

```
name: Bump version
on:
  push:
    branches:
      - master
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@master
    - name: Bump version and push tag
      uses: anothrNick/github-tag-action@1.7.1
      env:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
      id: bump_version
    - name: use Node.js 10.x
      uses: actions/setup-node@v1
    - name: Publish new version to npm
      run: |
         npm version $NEW_VERSION
         npm publish
      env:
         NPM_AUTH_TOKEN: ${{ secrets.NPM_AUTH_TOKEN }}
         NEW_VERSION: ${{ steps.bump_version.outputs.new_tag }}
```

I haven't tested this... so I'm not sure if it will actually work, but I _think_ it looks right. 

`Publish new version to npm` would update and publish the `new_tag` (output from `github-tag-action`) to npm using the `NPM_AUTH_TOKEN` setup in your secrets.

**Comment by anothrNick** [comment]
@foxmicha I've confirmed the following workflow works:

```
name: Bump version and publish to NPM
on:
  push:
    branches:
      - master
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@master
    - name: Bump version and push tag
      uses: anothrNick/github-tag-action@1.7.1
      env:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
      id: bump_version
    - name: use Node.js 10.x
      uses: actions/setup-node@v1
    - name: Publish new version to npm
      run: |
         echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" > ~/.npmrc
         npm version --no-git-tag-version $NEW_VERSION
         npm install
         npm publish
      env:
         NPM_TOKEN: ${{ secrets.NPM_AUTH_TOKEN }}
         NEW_VERSION: ${{ steps.bump_version.outputs.new_tag }}
```

I made a few changes to get the npm auth token to work in the action script, but this successfully publishes to npm with the new tag generated by `anothrNick/github-tag-action@1.7.1`.

**Comment by foxmicha** [comment]
Thanks for following up @anothrNick , this is pretty close to what I was hoping to do. A few caveats though:

1. I needed to push the new version back to master
    a. Master in this case is a protected branch, which appears to not be supported by the `GITHUB_TOKEN`
2. Disabling the protection and pushing anyway caused an endless workflow

It looks like there still some kinks to work out for automating the update process via GitHub actions. I appreciate your help, but I think I'm at the end of this road. Thanks!

**Comment by anothrNick** [comment]
Hmm, OK I guess I'm not sure I understand what you are trying yo do.... curious what you are doing that is causing an endless workflow loop.

**Comment by foxmicha** [comment]
The breakdown goes something like this:

1. Push to master - trigger workflow
2. Tag latest commit - push to tags
3. Perform `npm version --no-git-tag-version $NEW_VERSION`
4. Push to master - trigger workflow from within workflow
5. Newest commit isn't tagged - tag latest commit - push to tags

And on and on and on.

**Comment by anothrNick** [comment]
So you are doing something in your workflow that requires another push to master (Step 4)?

**Comment by foxmicha** [comment]
Yes - the new version associated with the `npm version` that is stored in the `package.json` needs to be pushed back into master.

**Comment by anothrNick** [comment]
I see, that was originally why I added `--no-git-tag-version`... so it wouldn't be committed. Although I understand why you'd want it in `package.json`, you then have 2 source of truths for the package version (albeit package.json is sourced by the tag action).

An option I've followed is to set the package version to `0.0.0` and just let the action and git dictate your tagged/released versions. If you don't want to do that you'll need an alternative approach.
